### ISSUE TITLE
spurious perf warning

### ISSUE THREAD

**Issue by zmike** [KhronosGroup/Vulkan-ValidationLayers#4199]
When blitting from an image to the same image (for mipmap generation) I get this perf warning:

```
UNASSIGNED-CoreValidation-DrawState-InvalidImageLayout(WARN / PERF): msgNum: 1303270965 - Validation Performance Warning: [ UNASSIGNED-CoreValidation-DrawState-InvalidImageLayout ] Object 0: handle = 0x7fcb54613580, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x4dae5635 | vkCmdBlitImage: For optimal performance VkImage 0x951c5d0000001f7f[] layout should be VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL instead of GENERAL.
```

Probably this shouldn't be emitted for the same image case because it's not an accurate recommendation.
